fighters/duckhunt/src/opff.rs

```diff
@@ -4,44 +4,130 @@ use super::*;
+        if fighter.motion_frame() <= 16.0 {
+            fighter.set_rate(2.0);
+        }
+        if fighter.motion_frame() > 16.0 {
+            fighter.set_rate(1.0);
+        }
+    }
+    if fighter.is_status(*FIGHTER_STATUS_KIND_SPECIAL_LW) {
+        if fighter.motion_frame() <= 6.0 {
+            fighter.set_rate(0.7);
+        }
+        if fighter.motion_frame() > 6.0 {
+            fighter.set_rate(0.9);
+        }
+    }
+}
 
-    // Frame Data
-    frame_data(boma, status_kind, motion_kind, frame);
+extern "Rust" {
+    fn gimmick_flash(boma: &mut BattleObjectModuleAccessor);
 }
 
-unsafe fn frame_data(boma: &mut BattleObjectModuleAccessor, status_kind: i32, motion_kind: u64, frame: f32) {
-    if status_kind == *FIGHTER_STATUS_KIND_SPECIAL_N {
-        if frame <= 16.0 {
-            MotionModule::set_rate(boma, 2.0);
-        }
-        if frame > 16.0 {
-            MotionModule::set_rate(boma, 1.0);
-        }
+unsafe fn gunman_timer(fighter: &mut L2CFighterCommon) {
+    let timer = VarModule::get_int(fighter.object(), vars::duckhunt::GUNMAN_TIMER);
+    if  timer != 0 {
+        VarModule::set_int(fighter.object(), vars::duckhunt::GUNMAN_TIMER, (timer-1));
+    }
+    if timer == 1 {
+        gimmick_flash(fighter);
     }
 }
 
 #[utils::macros::opff(FIGHTER_KIND_DUCKHUNT )]
 pub fn duckhunt_frame_wrapper(fighter: &mut smash::lua2cpp::L2CFighterCommon) {
     unsafe {
         common::opff::fighter_common_opff(fighter);
-		duckhunt_frame(fighter)
+        duck_jump_cancel(fighter);
+        frame_data(fighter);
+        gunman_timer(fighter);
     }
 }
 
-pub unsafe fn duckhunt_frame(fighter: &mut smash::lua2cpp::L2CFighterCommon) {
-    if let Some(info) = FrameInfo::update_and_get(fighter) {
-        moveset(&mut *info.boma, info.id, info.cat, info.status_kind, info.situation_kind, info.motion_kind.hash, info.stick_x, info.stick_y, info.facing, info.frame);
+#[smashline::weapon_frame_callback]
+pub fn gunman_callback(weapon: &mut smash::lua2cpp::L2CFighterBase) {
+    unsafe { 
+        if weapon.kind() != WEAPON_KIND_DUCKHUNT_GUNMAN {
+            return
+        }
+        let owner_id = WorkModule::get_int(weapon.module_accessor, *WEAPON_INSTANCE_WORK_ID_INT_LINK_OWNER) as u32;
+        if weapon.is_status(*WEAPON_DUCKHUNT_GUNMAN_STATUS_KIND_READY) {
+            let duckhunt = utils::util::get_battle_object_from_id(owner_id);
+            let duckhunt_boma = &mut *(*duckhunt).module_accessor;
+            if duckhunt_boma.is_cat_flag(Cat1::SpecialLw) && duckhunt_boma.is_button_trigger(Buttons::Special | Buttons::SpecialRaw) && WorkModule::get_int(weapon.module_accessor, *WEAPON_INSTANCE_WORK_ID_INT_LIFE) > 25 {
+                PLAY_STATUS(weapon, Hash40::new("se_duckhunt_special_l09"));
+                let gunman_kind = WorkModule::get_int(weapon.boma(), *WEAPON_DUCKHUNT_GUNMAN_INSTANCE_WORK_ID_KIND);
+                if PostureModule::lr(weapon.boma()) == -1.0 {
```

Comment:
instead of having a huge `if` here with duplicated code (I assume), you could just use the `lr` value in your `EFFECT_FOLLOW` arguments and it would be a lot simpler and cleaner.